src/tools/fuzzing/fuzzing.cpp

```diff
@@ -693,9 +693,77 @@ Expression* TranslateToFuzzReader::makeHangLimitCheck() {
 }
 
 Expression* TranslateToFuzzReader::makeLogging() {
-  auto type = getLoggableType();
-  return builder.makeCall(
-    std::string("log-") + type.toString(), {make(type)}, Type::none);
+  auto makeLoggingCall = [&](Expression* value) {
+    assert(isLoggableType(value->type));
+    return builder.makeCall(
+      std::string("log-") + value->type.toString(), {value}, Type::none);
+  };
+
+  // We may choose to log a local, if there are any.
+  auto* func = funcContext->func;
+  auto numLocals = func->getNumLocals();
+
+  auto choice = upTo(4);
+  if (choice < 2 || numLocals == 0) {
+    // 50% of the time, pick a loggable type and make something of that type
+    // to log. Also do so when there are no locals to log.
+    return makeLoggingCall(make(getLoggableType()));
+  }
+
+  // 50% of the time, pick a local and log it, either shallowly or deeply.
+  auto index = upTo(numLocals);
+  auto type = func->getLocalType(index);
+  if (isLoggableType(type)) {
+    // We can log this directly. Get it and log that.
+    return makeLoggingCall(builder.makeLocalGet(index, type));
+  }
+
+  if (type.isRef()) {
```

Comment:
How about inverting this condition to save a level of nesting on most of the following code?